pkg/agent/storage/gcs_test.go

```diff
@@ -0,0 +1,75 @@
+/*
+Copyright 2022 The KServe Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package storage
+
+import (
+	"fmt"
+	logger "log"
+	"os"
+	"path/filepath"
+	"testing"
+
+	gstorage "cloud.google.com/go/storage"
+	"github.com/onsi/gomega"
+)
+
+func TestDownloadModel(t *testing.T) {
+	g := gomega.NewGomegaWithT(t)
+	storageURIs := []string{
+		"gs://kfserving-examples/models/torchserve/image_classifier/v1",
+		"gs://kfserving-examples/models/torchserve/image_classifier/no-object",
+		"gs://bucket-not-exist/models/tensorflow/flowers",
```

Comment:
Can we use mocks instead of real gcs urls, ideally we'd like to run unit tests locally without depending on internet.